dev_guide/builds.adoc

```diff
@@ -191,6 +191,84 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended-builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might increase the size of the image or introduce vulnerabilities
+that can be exploited.
+
+To avoid these problems, source-to-image introduces a two image build process
+that allows an application to be built via the normal flow in a builder image
+but then injects the resulting application artifacts into a runtime-only image
+for execution.
+
+To offer flexibility in this process, source-to-image will execute an
+*_assemble-runtime_* script inside the runtime image that allows further
```

Comment:
@PI-Victor Our [docs guidelines ](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference) state that 

    *_this-format_* 
is reserved for filenames and directories. I was thinking of the script as an operation performed, which is why I suggest only using backticks per the guidelines. I will follow up on the other sections for consistency.